### PR TITLE
DVC-1886 tighten up tests and harden functionality

### PR DESCRIPTION
**Pull request by squarepegg**

### CODE REVIEW

File: android-client-sdk/src/main/java/com/devcycle/sdk/android/api/DVCClient.kt [hunk 1]
```diff
@@ -33,7 +33,7 @@ class DVCClient private constructor(
     eventsUrl: String,
 
     private val coroutineScope: CoroutineScope = MainScope(),
-    private val coroutineContext: CoroutineContext = Dispatchers.Default
+    coroutineContext: CoroutineContext = Dispatchers.Default
```
Comment (ajwootto) [inline android-client-sdk/src/main/java/com/devcycle/sdk/android/api/DVCClient.kt:36]: why this change?

Comment (squarepegg) [reply android-client-sdk/src/main/java/com/devcycle/sdk/android/api/DVCClient.kt]: that shouldn't be there, reverted.

**Review by ajwootto — APPROVED** [review]
